datafusion/optimizer/src/decorrelate_dependent_join.rs

```diff
@@ -299,25 +300,35 @@ impl DependentJoinDecorrelator {
             builder = builder.project(new_exprs)?;
         }
 
-        self.delim_scan_id = decorrelator.delim_scan_id;
         self.merge_child(&decorrelator);
         return builder.build();
     }
     fn merge_child(&mut self, child: &Self) {
         self.delim_scan_id = child.delim_scan_id;
-        for entry in child.correlated_column_to_delim_column.iter() {
-            self.correlated_column_to_delim_column
-                .insert(entry.0.clone(), entry.1.clone());
-        }
+        // TODO:
```

Comment:
remove this, so any delim_columns now respect the parent's delim column instead of the right child's delim columns